lib/puppet/provider/host/augeas.rb

```diff
@@ -57,14 +59,12 @@ def self.instances
 
   def self.prefetch(resources)
     targets = []
-    resources.each do |name, resource|
+    resources.each do |_name, resource|
       targets << target(resource) unless targets.include? target(resource)
     end
-    hosts = targets.inject([]) { |hosts,target| hosts += get_resources({:target => target}) }
+    hosts = targets.reduce([]) { |_acc, elem| hosts += get_resources({ target: elem }) }
```

Comment:
It's weird to use reduce but then don't use the accumulator and this is likely to introduce a bug. This is probably because it's shadowing the outer variable. I think this is what's intended.
```suggestion
    hosts = targets.reduce([]) { |acc, elem| acc += get_resources({ target: elem }) }
```

Reply:
It was done by rubocop auto_correct, will check your suggestion